document.write("\
<div class=\"w3-card-4 w3-margin w3-white\" style=\"border: solid 3px #954955;\">\
<div class=\"w3-row w3-padding w3-dark-gray\" style=\"width:100%\" width = \"200\"  >\
<div class=\"w3-container\"><p>\
<h3><b><!TITLE!></b></h3>\
<h5>3d Printing <span class=\"w3-opacity\"><!DATE!></span></h5>\
</div></div>\
<div class=\"w3-container\">\
<p>\
<!TEXT!>\
\
<div class=\"w3-row\">\
<div class=\"w3-col m8 s12\">\
<form method=\"get\" action=\"<!POST_NO!>">  <p><button class=\"w3-button w3-padding-large w3-white w3-border\"><b>READ MORE</b></button></p></form>\
</div>\
<p>\
</div>\
</div>\
</div>");